src/CMakeLists.txt

```diff
@@ -27,13 +27,18 @@ if(AMALGAMATION_BUILD)
   add_library(duckdb SHARED "${PROJECT_SOURCE_DIR}/src/amalgamation/duckdb.cpp")
   target_link_libraries(duckdb ${DUCKDB_SYSTEM_LIBS})
   link_threads(duckdb)
-  link_extension_libraries(duckdb)
+
+  if(${LINK_EXTENSIONS})
+    link_extension_libraries(duckdb)
```

Comment:
imo it would be a bit cleaner to check the flag inside the `link_extension_libraries` function itself, rather than before calling the function.

Reply:
Yes i thought about that but I thought that would propagate unnecessarily to the other binaries.
Will do!